p2p/simulations/adapters/inproc.go

```diff
@@ -356,3 +352,33 @@ func (api *SimAdminAPI) RemovePeer(url string) (bool, error) {
 	api.SimNode.RemovePeer(node)
 	return true, nil
 }
+
+// PeerEvents creates an RPC sunscription which receives peer events from the
+// underlying p2p.Server
+func (api *SimAdminAPI) PeerEvents(ctx context.Context) (*rpc.Subscription, error) {
+	notifier, supported := rpc.NotifierFromContext(ctx)
+	if !supported {
+		return &rpc.Subscription{}, rpc.ErrNotificationsUnsupported
+	}
+
+	rpcSub := notifier.CreateSubscription()
+
+	go func() {
+		events := make(chan *p2p.PeerEvent)
+		sub := api.SubscribeEvents(events)
+		defer sub.Unsubscribe()
+
+		for {
+			select {
+			case event := <-events:
+				notifier.Notify(rpcSub.ID, event)
+			case <-rpcSub.Err():
+				return
+			case <-notifier.Closed():
```

Comment:
I think you may need a 4th case here to catch `sub.Err()` or something to quit if the sub errors no?